### ISSUE TITLE
Publish JAR to mvncentral?

### ISSUE THREAD

**Issue by esend7881** [Juniper/netconf-java#31]
Hi All,
I am planning to use your Netconf API in my project and I was wondering if you'd consider publishing your JAR to mvncentral?

http://maven.apache.org/repository/guide-central-repository-upload.html
https://central.sonatype.org/pages/requirements.html

I can help, however there are a few requirements, such as filling out a licence section in the POM and creating a javadoc/source jar file.

Appreciate all your hard work!

**Comment by esend7881** [comment]
Added this commit to help towards this: https://github.com/Juniper/netconf-java/pull/30/commits/448c12e4a7d2ec0bfc7b247e02c5d058729e7c8f

**Comment by peterjhill** [comment]
looks good. I definitely agree that having this published up to mvncentral would be great. Then I can stop publishing to our internal mvn.

**Comment by esend7881** [comment]
> looks good. I definitely agree that having this published up to mvncentral would be great. Then I can stop publishing to our internal mvn.

I agree, especially since the source code is on github anyway, might as well take the next step.

**Comment by ydnath** [comment]
Thanks Eric @esend7881 

I have merged this. However, I am not aware of Maven publishing process. I will go through the documentation and get back.

**Comment by esend7881** [comment]
> Thanks Eric @esend7881
> 
> I have merged this. However, I am not aware of Maven publishing process. I will go through the documentation and get back.

FYI - I published your code to my companies mvn internal server, so I am good now. But yeah it would be great to have it on mvn central for others out there who'd want to use it (more easily).

**Comment by peterjhill** [comment]
github has this: https://docs.github.com/en/actions/guides/publishing-java-packages-with-maven
That has info on how to use Sonatypes open source maven central artifact repository (to host the jar files).

This requires opening a jira account with sonatype:
https://central.sonatype.org/publish/

This then give whomever sets this up permission to create releases. We then add some foo to the pom file and some github secrets for the account integration between github and sonatype. Details are all in there.

If you want to set up a sonatype account and add me as a person who can set up the integration I can do that. Not sure how busy you are. :)

**Comment by esend7881** [comment]
@peterjhill I found some time and created this jira ticket https://issues.sonatype.org/browse/OSSRH-69987

Also I made a branch in my forked repo that has updates to the POM file https://github.com/esend7881/netconf-java/tree/RELEASE/2.1.1.4

We'll see what happens with the JIRA ticket at this point.

**Comment by peterjhill** [comment]
I pinged our friends at Juniper and asked them to comment in that jira ticket.

Thanks Erik!

**Comment by GregDThomas** [comment]
Hi folks - any chance of getting this looked at - would make things easier! Thanks.

**Comment by ydnath** [comment]
Let me review the process and get this done.

**Comment by dkowis** [comment]
I guess this got lost? It would still be nice to use maven central.

However, until then, you can use jitpack, if you're allowed to pull jar files from somewhere other than maven central:

https://jitpack.io/#Juniper/netconf-java
